src/note/commitment.rs

```diff
@@ -140,3 +147,40 @@ impl PartialEq for ExtractedNoteCommitment {
 }
 
 impl Eq for ExtractedNoteCommitment {}
+
+#[cfg(test)]
+mod tests {
+    use crate::constants::fixed_bases::{
+        NOTE_COMMITMENT_PERSONALIZATION, NOTE_ZSA_COMMITMENT_PERSONALIZATION,
+    };
+    use crate::note::commitment::NoteCommitTrapdoor;
+    use ff::Field;
+    use halo2_gadgets::sinsemilla::primitives as sinsemilla;
+    use pasta_curves::pallas;
+    use rand::{rngs::OsRng, Rng};
+
+    #[test]
+    fn test_note_commit() {
```

Comment:
As previously mentioned, this does not test `derive()`. This seems to be a lower-level test, am I missing something?

Reply:
I renamed this test